src/Backups/BackupImpl.cpp

```diff
@@ -834,7 +836,7 @@ void BackupImpl::writeFile(const BackupFileInfo & info, BackupEntryPtr entry)
 
     /// We need to copy whole file without archive, we can do it faster
     /// if source and destination are compatible
-    if (!use_archive && writer->supportNativeCopy(reader_description))
+    if (!use_archive && !context->getBackupsThrottler() && writer->supportNativeCopy(reader_description))
```

Comment:
> Instead, we use some other kind of resource somewhere else (e.g. maybe in the cloud for an S3 copy).

Not always, sometimes it is plain copy on the disk, and in this case we should apply throttling, but indeed copy from S3 to S3 does not make sense to throttle, so I will improve this part.

>You can instead do throttler->add(file_size) here and be done

Throttling should be done for a small batches otherwise it is useless, if throttling will be applied only here (`throttle->add(file_size)`) then it will simply copy the whole file (that can be 10GB+, which can take 102 seconds, with average 100MB/s read speed) and the sleep very long time, while during it will copy the file (102 seconds in this example) production queries will be affected

Reply:
Yes, that's why I said this can only be done if you do not want to protect the resource, but looks like it is indeed what you need. This is again pointless for S3 to S3 copy, so let's just skip throttling in this case.

Reply:
Fixed in 1c86adeeec9f1a215e1e06a2ee1ac1d56dc9c2be